display/man.rs

```diff
@@ -0,0 +1,445 @@
+///
+/// # Returns
+///
+/// [Vec<u8>] STDOUT of called formatter.
+///
+/// # Errors
+///
+/// [ManError] if failed to execute formatter.
+fn format_man_page(man_page: Vec<u8>) -> Result<Vec<u8>, ManError> {
+    let width = get_page_width()?;
+
+    let formatters = [groff_format, nroff_format, mandoc_format];
+
+    for formatter in &formatters {
+        match formatter(&man_page, width) {
+            Ok(formatted_man_page) => return Ok(formatted_man_page),
+            Err(err) if err.kind() == io::ErrorKind::NotFound => continue,
+            Err(err) => return Err(err.into()),
+        }
+    }
+
+    Err(ManError(
+        "neither groff(1), nor nroff(1), nor mandoc(1) are installed".to_string(),
+    ))
+}
+
+/// Formats man page content into appropriate format.
+///
+/// # Arguments
+///
+/// `man_page` - [Vec<u8>] with content that needs to displayed.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [std::io::Error] if failed to execute pager or failed write to its STDIN.
+fn display_pager(man_page: Vec<u8>) -> Result<(), io::Error> {
+    let pager = std::env::var("PAGER").unwrap_or_else(|_| "more".to_string());
+
+    let args = if pager.ends_with("more") {
+        vec!["-s"]
+    } else {
+        vec![]
+    };
+
+    spawn(&pager, args, Some(&man_page), Stdio::inherit())?;
+
+    Ok(())
+}
+
+/// Displays man page
+///
+/// # Arguments
+///
+/// `name` - [str] name of system documentation.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [ManError] if man page not found, or any display error happened.
+fn display_man_page(name: &str) -> Result<(), ManError> {
+    let cat_output = get_man_page(name)?;
+    let formatter_output = format_man_page(cat_output)?;
+    display_pager(formatter_output)?;
+
+    Ok(())
+}
+
+/// Displays man page summaries for the given keyword.
+///
+/// # Arguments
+///
+/// `keyword` - [str] name of keyword.
+///
+/// # Returns
+///
+/// Nothing
+///
+/// # Errors
+///
+/// [ManError] if call of `apropros` utility failed.
+fn display_summary_database(keyword: &str) -> Result<(), ManError> {
+    let output: Output = Command::new("apropos").arg(keyword).output()?;
+
+    if !output.status.success() {
+        return Err(ManError("apropos command failed".to_string()));
+    }
+
+    let result = String::from_utf8_lossy(&output.stdout);
+    print!("{result}");
+
+    Ok(())
+}
+
+/// Main function that handles the program logic. It processes the input
+/// arguments, and either displays man pages or searches the summary database.
+///
+/// # Arguments
+///
+/// `args` - [Args] set of incoming arguments.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [ManError] wrapper of program error.
+fn man(args: Args) -> Result<(), ManError> {
+    let any_path_exists = MAN_PATHS.iter().any(|path| PathBuf::from(path).exists());
+
+    if !any_path_exists {
+        return Err(ManError("man paths to man pages doesn't exist".to_string()));
```

Comment:
What's the point in having a custom error type if it's just a wrapper over a String? Make a proper error type